packages/block-editor/src/components/block-mover/index.js

```diff
@@ -44,24 +44,60 @@ export class BlockMover extends Component {
 	}
 
 	render() {
-		const { onMoveUp, onMoveDown, isFirst, isLast, isDraggable, onDragStart, onDragEnd, clientIds, blockElementId, blockType, firstIndex, isLocked, instanceId, isHidden, rootClientId } = this.props;
+		const { onMoveUp, onMoveDown, orientation, isRTL, isFirst, isLast, isDraggable, onDragStart, onDragEnd, clientIds, blockElementId, blockType, firstIndex, isLocked, instanceId, isHidden, rootClientId } = this.props;
 		const { isFocused } = this.state;
 		const blocksCount = castArray( clientIds ).length;
 		if ( isLocked || ( isFirst && isLast && ! rootClientId ) ) {
 			return null;
 		}
 
+		const getArrowIcon = ( hdir ) => {
+			let arrow;
+			switch ( hdir ) {
+				case 'left':
+					arrow = isRTL ? rightArrow : leftArrow;
+					arrow = orientation === 'horizontal' ? arrow : upArrow;
+					break;
+
+				case 'right':
+					arrow = isRTL ? leftArrow : rightArrow;
+					arrow = orientation === 'horizontal' ? arrow : downArrow;
+					break;
+
+				default:
+					break;
+			}
+			return arrow;
+		};
+
+		const getMovementDirection = ( mdir ) => {
+			switch ( mdir ) {
+				case 'left':
+					mdir = isRTL ? 'right' : mdir;
+					break;
+
+				case 'right':
+					mdir = isRTL ? 'left' : mdir;
+					break;
+
+				default:
+					break;
+			}
+			return mdir;
+		};
+
 		// We emulate a disabled state because forcefully applying the `disabled`
 		// attribute on the button while it has focus causes the screen to change
 		// to an unfocused state (body as active element) without firing blur on,
 		// the rendering parent, leaving it unable to react to focus out.
 		return (
-			<div className={ classnames( 'editor-block-mover block-editor-block-mover', { 'is-visible': isFocused || ! isHidden } ) }>
+			<div className={ classnames( 'editor-block-mover block-editor-block-mover', { 'is-visible': isFocused || ! isHidden, 'is-horizontal': orientation === 'horizontal' } ) }>
 				<IconButton
 					className="editor-block-mover__control block-editor-block-mover__control"
 					onClick={ isFirst ? null : onMoveUp }
-					icon={ upArrow }
-					label={ __( 'Move up' ) }
+					icon={ getArrowIcon( 'left' ) }
+					// translators: %s: Horizontal direction of block movement ( left, right )
+					label={ orientation === 'horizontal' ? sprintf( __( 'Move %s' ), getMovementDirection( 'left' ) ) : __( 'Move up' ) }
```

Comment:
Could we move the `orientation === 'horizontal'` check into `getMovementDirection` so that there's' slightly less repetition and so that `getMovementDirection` and `getArrowIcon` behave similarly?

Reply:
yes, reads much better now, thank you :)